File: sdk/dotnet/src/Experimently/Exceptions/ExperimentationException.cs
namespace Experimently.Exceptions;

/// <summary>
/// Base exception for all Experimently SDK errors.
/// </summary>
public class ExperimentationException : Exception
{
    /// <summary>Creates the exception with a message.</summary>
    public ExperimentationException(string message) : base(message) { }

    /// <summary>Creates the exception with a message and an underlying cause.</summary>
    public ExperimentationException(string message, Exception innerException)
        : base(message, innerException) { }
}

File: sdk/php/src/Errors/ExperimentationException.php
<?php

declare(strict_types=1);

namespace Experimently\Errors;

/**
 * Base exception for all Experimently SDK errors.
 */
class ExperimentationException extends \RuntimeException
{
    public function __construct(string $message = '', int $code = 0, ?\Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}
